### PR TITLE
use refresh() instead of reassigning href property

### LINKED ISSUE #22113: JS Page reload not working when there's hash in URL
**Describe the bug**
When the cart updates, the JS code in `cart.js` specifies how to handle the updates.
If the cart is empty, and the checkout element is on the same page as the cart page, the code should refresh the page.
See:

https://github.com/woocommerce/woocommerce/blob/d0491072e8a7d0a4535db0ec2bae4a6d43ff64fd/assets/js/frontend/cart.js#L73-L76

The problem with this code,  is that the line 
`window.location.href = window.location.href;`
only works when there's no hash on the page URL.
For example, when the URL is `http://domain.com/checkout#cart`, the refresh with href=href wouldn't work.
A better way to refresh would be to call `window.location.refresh()`.

### PR DESCRIPTION
**Pull request by rrennick**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Use window refresh() instead of reassigning href property.

Closes #22113 .

### How to test the changes in this Pull Request:

1. Add a product to the cart.
2. View the cart.
3. Remove the item from the cart.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### Changelog entry

> Reload the cart page when the cart is empty.

### CODE REVIEW

File: assets/js/frontend/cart.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ jQuery( function( $ ) {
 
 		// No form, cannot do this.
 		if ( $( '.woocommerce-cart-form' ).length === 0 ) {
-			window.location.href = window.location.href;
+			window.location.refresh();
```
Comment (joshuatf) [inline assets/js/frontend/cart.js]: I think we might want to use `window.location.reload()` here as `refresh()` is not a function.

**Review by joshuatf — COMMENTED** [review]
Thanks for this PR @rrennick !  This seems like a good idea- just a couple quick notes:
* Made a comment below about refresh/reload function
* Should we also change the other `window.location.href` assignment below?  https://github.com/woocommerce/woocommerce/blob/f13a0bc9eb49dad87b93e5b3ac019838e43b1404/assets/js/frontend/cart.js#L86
* For anyone else testing, in order to test I also had to click "Undo" after removing from cart to test in my environment.

File: assets/js/frontend/cart.js (hunk 1 above)
Comment (rrennick) [reply assets/js/frontend/cart.js]: Good catch, thanks. It's been fixed

**Comment by rrennick** [comment]
> Should we also change the other window.location.href assignment below?

@joshuatf Added this to the PR, thanks.

**Review by mikejolley — APPROVED** [review]

**Pull request merged**